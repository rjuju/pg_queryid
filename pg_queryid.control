# pg_queryid extension
comment = 'External query fingerprinting module'
default_version = '0.0.1'
module_pathname = '$libdir/pg_queryid'
relocatable = true
