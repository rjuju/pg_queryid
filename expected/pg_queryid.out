SHOW shared_preload_libraries;
   shared_preload_libraries    
-------------------------------
 pg_stat_statements,pg_queryid
(1 row)

SHOW compute_query_id;
 compute_query_id 
------------------
 off
(1 row)

CREATE EXTENSION pg_stat_statements;
CREATE SCHEMA ns1;
CREATE TABLE ns1.tbl1 AS SELECT 'ns1' AS val;
CREATE SCHEMA ns2;
CREATE TABLE ns2.tbl1 AS SELECT 'ns2' AS val;
CREATE TEMPORARY TABLE tmptbl(val text);
SET pg_queryid.use_object_names = on;
SET pg_queryid.ignore_schema = on;
SET pg_queryid.ignore_temp_tables = on;
SELECT * FROM pg_stat_statements_reset();
 pg_stat_statements_reset 
--------------------------
 
(1 row)

SET search_path to ns1;
SELECT * FROM tbl1;
 val 
-----
 ns1
(1 row)

SET search_path to ns2;
SELECT * FROM tbl1;
 val 
-----
 ns2
(1 row)

RESET search_path;
SELECT * FROM tmptbl;
 val 
-----
(0 rows)

SELECT queryid, query, calls FROM public.pg_stat_statements WHERE query LIKE '%tbl%';
       queryid       |       query        | calls 
---------------------+--------------------+-------
 8597862845229905026 | SELECT * FROM tbl1 |     2
(1 row)

SELECT * FROM pg_stat_statements_reset();
 pg_stat_statements_reset 
--------------------------
 
(1 row)

SET pg_queryid.ignore_temp_tables = off;
SELECT * FROM tmptbl;
 val 
-----
(0 rows)

SELECT queryid, query, calls FROM public.pg_stat_statements WHERE query LIKE '%tbl%';
       queryid       |        query         | calls 
---------------------+----------------------+-------
 1537637248015716743 | SELECT * FROM tmptbl |     1
(1 row)

